ui/src/components/dialog/__tests__/QDialog.cy.js

```diff
@@ -1,21 +1,131 @@
+import DialogWrapper from './DialogWrapper.vue'
+import { nextTick, ref } from 'vue'
+import { vModelAdapter } from '@quasar/quasar-app-extension-testing-e2e-cypress'
+
+function mountQDialogWrapper (options) {
+  return cy.mount(DialogWrapper, options)
+}
+
+function getHostElement () {
+  return cy.dataCy('dialog-card')
+}
+
+function closeDialogViaBackdrop () {
+  return cy.get('.q-dialog__backdrop').click({ force: true })
+}
+
+// TODO: this only works in `withinDialog` context,
```

Comment:
Check out this TODO
If you find a way to make `closeDialogViaEscKey` work both inside and outside a `within` context, that would be cool

I tried with `cy.window()` and `cy.document()` already, which I recalled worked fine, but it seems like they changed that

Reply:
Could not find a better way to handle this either.

Reply:
Ok, let's keep the TODO and see if we are able to spend some more time on this in a follow up PR

Reply:
Documented these new helpers and their limitations here: https://github.com/quasarframework/quasar-testing/issues/355